### PR TITLE
SI/DeviceGBA: Improve link stability

### PR DESCRIPTION
**Pull request by endrift**
Building on #9550 I found some more issues that could cause the link to get out of sync.

The original flush technique didn't work properly; it didn't work before the first RESET was sent, which generally meant the first RESET would be junk too. This replaces it with flushing _before_ sending any message. This approach isn't perfect but clears out the junk much more reliably.

This also adds a minor wait for the "non-booted" state (polling for status instead of actual data transfer) to improve the stability further.

### CODE REVIEW

**Comment by JMC47** [comment]
Fixes connecting to Final Fantasy: Crystal Chronicles.

File: Source/Core/Core/HW/SI/SI_DeviceGBA.cpp [hunk 1]
```diff
@@ -258,12 +261,12 @@ int GBASockServer::Receive(u8* si_buffer, u8 bytes)
   if (!m_client)
     return 0;
 
+  sf::SocketSelector selector;
+  selector.add(*m_client);
   if (m_booted)
```
Comment (BhaaLseN) [inline Source/Core/Core/HW/SI/SI_DeviceGBA.cpp:266]: I wonder if its worth documenting why `m_booted` affects how long we can/should wait for a socket, and if those numbers have any significance (vs. being guesses that just work in practise)

Comment (endrift) [reply Source/Core/Core/HW/SI/SI_DeviceGBA.cpp]: Apparently the number didn't even work on Windows, so I replaced it with redefining what `m_booted` represents.

**Comment by JMC47** [comment]
This does improve linking with Final Fantasy: Crystal Chronicles.  I do not know of any regressions.

**Review by BhaaLseN — APPROVED** [review]
Code looks good, but untested. Plus, I don't know enough of the protocol to say whether there's other things looming that might pop out later.

**Comment by endrift** [comment]
I know I've been testing it and I think @JMC47 has been too, so "untested" might not be the most accurate description here.

**Comment by BhaaLseN** [comment]
Untested by _me_ ;)

**Pull request merged**